### PR TITLE
`PositionBodyComponent` to have optional arguments

### PR DESCRIPTION
**Pull request by spydon**
# Description

Now initialization of the `PositionComponent` can be done in `PositionBodyComponent.onLoad`, making it cleaner when extending it (which you are forced to do anyways since you'll have to implement `createBody`).

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `flutter format` and the `flutter analyze` does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [ ] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

**Comment by spydon** [comment]
Fails analyze due to #1120 not being merged yet.

File: packages/flame_forge2d/example/lib/camera_sample.dart [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 
 import 'domino_sample.dart';
+import 'sprite_body_sample.dart';
 
 class CameraSample extends DominoSample {
   @override
   void onTapDown(TapDownInfo details) {
     final position = details.eventPosition.game;
-    final pizza = Pizza(position, pizzaImage);
+    final pizza = Pizza(position);
```
Comment (erickzanardo) [inline packages/flame_forge2d/example/lib/camera_sample.dart:10]: Kawabunga!

**Review by erickzanardo — APPROVED** [review]

File: packages/flame_forge2d/lib/position_body_component.dart [hunk 2]
```diff
@@ -7,49 +7,61 @@ import 'forge2d_game.dart';
...
 
-  void updatePositionComponent() {
-    positionComponent.position.setFrom(body.position);
-    positionComponent.position.y *= -1;
-    positionComponent
-      ..angle = -angle
-      ..size = size;
+  void _updatePositionComponent() {
+    positionComponent?.position.setFrom(body.position);
+    positionComponent?.position.y *= -1;
```
Comment (st-pasha) [inline packages/flame_forge2d/lib/position_body_component.dart:64]: `!`

Comment (spydon) [reply packages/flame_forge2d/lib/position_body_component.dart:64]: Shouldn't really matter here? Is one of them a cheaper operation than the other?

Comment (spydon) [reply packages/flame_forge2d/lib/position_body_component.dart]: Fixed it!

**Pull request merged**